ONSdigital/eq-author-app
# EAR-551 Sort out destination picker

## Useful links
[Ticket](https://collaborate2.ons.gov.uk/jira/browse/EAR-551)
[Designs](https://overflow.io/s/96LRCTYQ?node=2d584288&prototype)

### What is the context of this PR?
Sorting out the destination content picker.

It should:
- Allow users to pick questions from the same sections
- Provide only the sections if they are routing outside of the current section
- Provide other destinations if any of the above don't work

### How to review
- Create questionnaire
- Add four question pages in section one
- Add an additional 3 sections
- Go to S1 q2 and create a routing rule
- Click `select a destination` and the modal should display
NOTE: Design of modal being fixed in another ticket
- Should be three tabs on the left (`Current`, `Later`, `Other`)
- Each tab should display correct destination values
- If clicking the `Else` destination selector, `Later` should not appear
NOTE: should be able to switch tabs and choose a destination with `Space` or `Enter`
- Confirm the selection and check it appears in the destination picker
